### ISSUE TITLE
Pull-request - Don't throw error if there are no changes

### ISSUE THREAD

**Issue by Shaun-Harrison** [github/hub#2645]
**The problem I'm trying to solve:**
I want to use this function on a schedule to create PR between branches automatically, there may not always be changes to merge from branch to branch

**How I imagine hub could expose this functionality:**
Currently the action will fail if there is no changes when trying to create the PR, ideally there could be a flag which would stop it failing , for the time being I can just have other actions run even if there's a failure, but I don't want an overall failure status in github action

**Comment by mislav** [comment]
Hi, thank you for explaining your use-case.

If both branches are in the same repository, is it possible that you add a check to your script before running `hub pull-request`? Something like

```sh
if ! git diff --quiet "${base_branch}...${head_branch}"; then
  hub pull-request ...
fi
```

This will avoid calling `hub pull-request` unless there are differences between base branch and head branch.

The reason I suggest this is that I think it might be easier for you to implement this check than waiting for hub to somehow implement this.

**Comment by Shaun-Harrison** [comment]
Hi @mislav 

I did try going down this route but various outputs , more than likely something I was doing wrong! 

However a colleague has provided me with a solution, we have added || `echo "No changes"` to the end of the hub command which works quite nicely! 
![image](https://user-images.githubusercontent.com/46562010/98680634-95f73600-2359-11eb-81ae-d7dd96e2fa44.png)

The action is not then failing!
